### DISCUSSION TITLE
Multiple Filters and multiple columns

### DISCUSSION THREAD

**Discussion by Sebastiangperez** [sakanjo/laravel-easy-metrics#6] [General]
There is a way that you can have multiple columns i mean , like most product sold by month but you can have multiple columns in one month or a range date ? 
Also , can you filter by month and product ?
What about nested , like , brand -> product by month ? 
Edit 
I have something like this using chart js and laravel , how can mimic the same with your library ? 

$Salespermonthchart = SalesperDealerShip::all(); // this is a view that came from a database grouping dealerships quantity of sales and for month 

        $dealershipsChart = $Salespermonthchart->groupBy('dealership');
        $groupedDate = $Salespermonthchart->groupBy('date')->keys()->toArray();
        $Chartdata['labels'] = $groupedDate;
        $i = 0;

        foreach ($dealershipsChart as $key => $value) {
            $Chartdata['datasets'][$i]['label'] = $key;
            if (array_key_exists($key, $this->DealershipColors)) { // Just an array with colors for each dealership
                $Chartdata['datasets'][$i]['backgroundColor'] = [$this->DealershipColors[$key]];
            }
            for ($j = 0; $j < count($groupedDate); $j++) {
                $Chartdata['datasets'][$i]['data'][$j] = 0;
                foreach ($value as $k => $v) {
                    if ($groupedDate[$j] == $v->date) {
                        $Chartdata['datasets'][$i]['data'][$j] = $v->Quantity;
                    }
                }
            }
            $i++;
        }

        return json_encode($Chartdata);

Also in the chart i have a min and max range date to select , full date.

**Comment by sakanjo** [comment]
@Sebastiangperez you can use multiple `Trend` classes for multiple columns,

For filtering you can filter by range as well as by model e.g. `->range(Range::MTD)` or by model `->modifyQuery(fn (Builder $query) => query->where('id', request(..))`.

**Reply by Sebastiangperez** [reply]
Nice, thanks for your response. 
Question , do you know how can achieve my code or adapt it to your library ? do you have an example of it ?

**Comment by sakanjo** [comment]
One of my projects metric page:

```php
<?php

namespace App\Filament\Widgets\Admin;

use App\Models;
use Arr;
use Filament\Widgets\StatsOverviewWidget as BaseWidget;
use Filament\Widgets\StatsOverviewWidget\Stat;
use SaKanjo\EasyMetrics\Metrics\Trend;

class CountOverview extends BaseWidget
{
    protected static array $models = [
        Models\Admin::class,
        Models\Manager::class,
        Models\User::class,
        // ...
    ];

    protected function getStats(): array
    {

        return Arr::map(
            static::$model,
            fn (string $model) => Stat::make(class_basename($model), $model::count())
                ->chart(Trend::make($model)->countByMonths()->getData())
                ->color('primary')
        );
    }
}
```
